common/lanelet2_extension/lib/CarmaTrafficSignal.cpp

```diff
@@ -60,6 +60,42 @@ LineStrings3d CarmaTrafficSignal::stopLine()
   return getParameters<LineString3d>(RoleName::RefLine);
 }
 
+Optional<ConstLineString3d> CarmaTrafficSignal::getStopLine(const ConstLanelet& llt) const 
+{
+  auto sl = stopLine();
+  if (sl.empty()) {
+    return {};
+  }
+  auto llts = getControlStartLanelets();
+  auto it = std::find(llts.begin(), llts.end(), llt);
+  if (llts.empty())
+  {
+    return {};
+  }
+  if (it == llts.end()) {
+    return {};
+  }
+  return sl.at(size_t(std::distance(llts.begin(), it)));
```

Comment:
This looks like an integer cast from an distance value which is being used to index into the linestring. 

This seems risky, we don't have a guarantee that linestrings in the map have fixed spacing between points. 

Am I missing something?